packages/flutter/lib/src/widgets/container.dart

```diff
@@ -105,6 +109,54 @@ class DecoratedBox extends SingleChildRenderObjectWidget {
   }
 }
 
+class _DynamicBox extends StatelessWidget {
+  /// This class determines which widget to build based on config's type.
+  const _DynamicBox(this.config, {required this.child});
+
+  final dynamic config;
+  final Widget child;
+
+  Never error(String description) => throw Exception('$description: $config');
```

Comment:
Should this be a FlutterError (the error message probably needs more details to be actionable for devlopers)? Or should the call sides really just be asserts?